LPHK.py

```diff
@@ -88,51 +98,74 @@ def init():
 
     ap = argparse.ArgumentParser()                             # argparse makes argument processing easy
     ap.add_argument(                                           # reimnplementation of debug (-d or --debug)
-        "-d", "--debug", 
-        help = "turn on debugging mode", action="store_true")
-    ap.add_argument(                                           # new option to automatically load a layout
-        "-l", "--layout", 
-        help = "load a layout", 
+        "-d", "--debug",
+        help = "Turn on debugging mode", action="store_true")
+    ap.add_argument(                                           # option to automatically load a layout
+        "-l", "--layout",
+        help = "Load an initial layout",
         type=argparse.FileType('r'))
-    ap.add_argument(                                           # new option to start minimised
-        "-m", "--minimised", 
+    ap.add_argument(                                           # option to start minimised
+        "-m", "--minimised",
         help = "Start the application minimised", action="store_true")
-
-    window.ARGS = vars(ap.parse_args())                        # store the arguments in a place anything can get to
-
-    if window.ARGS['debug']:
+    ap.add_argument(                                           # option to start without connecting to a Launchpad
+        "-s", "--standalone",
+        help = "Operate without connection to Launchpad", type=str, choices=[LP_MK1, LP_MK2, LP_MINI, LP_PRO])
+    ap.add_argument(                                           # option to start with launchpad window in a particular mode
+        "-M", "--mode",
+        help = "Launchpad mode", type=str, choices=[LM_EDIT, LM_MOVE, LM_SWAP, LM_COPY, LM_DEL, LM_RUN], default=LM_EDIT)
```

Comment:
It starts the program in the specified mode.  The default is starting it up in edit mode, so that clicking on a button will start the editor.  However, if you're operating without a launchpad, it makes more sense to start in run mode, so that clicking on a button will execute the macro.  I always start mine in run mode now.  The other modes are there just so you can chose one of them if you have a reason to,